fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -994,9 +1004,19 @@ where
         self.memory.write(self.owner, dst_addr, length)?;
         self.input_contracts.check(&contract_id)?;
 
-        let contract = super::contract::contract(self.storage, &contract_id)?;
-        let contract_bytes = contract.as_ref().as_ref();
-        let contract_len = contract_bytes.len();
+        let contract_len = contract_size(self.storage, &contract_id)?;
+
+        let mut contract_buffer: Vec<u8> = alloc::vec![0u8; contract_len as usize];
+        self.storage
+            .read_contract(&contract_id, &mut contract_buffer)
+            .transpose()
+            .ok_or(PanicReason::ContractNotFound)?
+            .map_err(RuntimeError::Storage)?;
+
+        if contract_buffer.len() != contract_len as usize {
+            return Err(PanicReason::ContractMismatch.into())
+        }
```

Comment:
This code is repeated. How about breaking it out to a helper function?

Reply:
Good shout: bd00e21fb

Reply:
Nice, thanks for updating!